orttraining/orttraining/test/graph/gradient_graph_builder_test.cc

```diff
@@ -798,6 +800,156 @@ class PipelineSplitter {
   }
 };
 
+// TODO: move to a proper location for pipeline training
+struct PipelineBatchInfo {
+  std::vector<std::pair<int64_t, int64_t>> events;
+  int64_t retired_batch;  // index of retired batch, so its data could be reused
+};
+
+class PipelineBatchPlanner {
```

Comment:
>PipelineBatchPlanner [](start = 6, length = 20)

I wonder if we want to setup the code structure for supporting multiple type of Planner in this PR ? 

e.g. a base Planner class, and this particular GenerateTimeline() impl is for OneForwadOneBackward schedule